localization/yabloc/yabloc_monitor/src/availability_module.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 TIER IV
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "availability_module.hpp"
+
+#include <rclcpp/logging.hpp>
+
+#include <memory>
+
+using PoseStamped = geometry_msgs::msg::PoseStamped;
+
+AvailabilityModule::AvailabilityModule(rclcpp::Node * node)
+: clock_(node->get_clock()),
+  latest_yabloc_pose_stamp_ptr_(nullptr),
+  timestamp_threshold_(node->declare_parameter<double>("availability/timestamp_tolerance"))
+{
+  sub_yabloc_pose_ = node->create_subscription<PoseStamped>(
+    "~/input/yabloc_pose", 10, [this](PoseStamped::ConstSharedPtr msg) { on_yabloc_pose(msg); });
+}
+
+bool AvailabilityModule::is_available() const
+{
+  if (latest_yabloc_pose_stamp_ptr_ == nullptr) {
+    return false;
+  }
+
+  const auto now = clock_->now();
+
+  const auto diff_pose = now - *latest_yabloc_pose_stamp_ptr_;
```

Comment:
Since this is a comparison of timestamps, how about renaming it or the following?

```c++
const auto diff_pose_sec = (now - *latest_yabloc_pose_stamp_ptr_).seconds();
```

Reply:
Thank you for the review :pray: 
Resolved: https://github.com/autowarefoundation/autoware.universe/pull/4395/commits/ad063e9960f27f4422a6806f2df460298abd03d2